### PR TITLE
Nc news 4

### PR DESCRIPTION
**Pull request by denton909**
HI 
I have completed task 4. Please could someone review this for me.

Thanks

Phil

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -36,14 +36,44 @@ describe("nc-news-3", () => {
...
+    describe('Error Handling', ()=>{
+        test('400 Bad Request. User inputs an id which is not a number this will return a 400 code and an error message', ()=>{
+         return request(app)
+        .get("/api/articles/mitch")
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Bad Request Invalid Input");
+        })
+        })
+        test('413 Request Entity Too Large. User inputs an id which is a number but is bigger than current amount of data this will return a 413 code and an error message', ()=>{
```
Comment (EddTheDuck) [inline __tests__/app.test.js]: I'm glad to see you removed this test

File: app.js [hunk 2]
```diff
@@ -1,10 +1,12 @@
 const { getTopics, getEndPoints, getArticlesById  } = require("./controller");
-const express = require('express')
+const express = require('express');
+const { invalidEndPoint } = require("./error-handling");
 const app = express();
 
 app.get('/api/topics', getTopics)
 app.get('/api', getEndPoints)
 app.get('/api/articles/:article_id',getArticlesById)
+app.use(invalidEndPoint)
```
Comment (EddTheDuck) [inline app.js:9]: the error handlers should be split by functionality (eg handleCustomErrors, handlePsqlErrors) instead of invalidEndPoint

**Review by EddTheDuck — CHANGES_REQUESTED** [review]
You need to add a more robust test and test for the properties of the objects you are returning. Once you have done that you can push and merge.

Try find a matcher that checks an object

File: app.js (hunk 2 above)
Comment (denton909) [reply app.js:9]: Updated this and changed name to handleCustomErrors

File: __tests__/app.test.js [hunk 3]
```diff
@@ -36,14 +36,44 @@ describe("nc-news-3", () => {
...
+           expect(body.article.article_id).toBe(1)
         })
     })
+    test("200 GET the article object that corresponds with the id number of 4 in the URL", () => {
+        return request(app)
+        .get("/api/articles/4")
+        .expect(200)
+        .then(({ body }) => {
+           expect(typeof body.article).toBe('object');
+           expect(body.article.article_id).toBe(4)
```
[the remark this answers is not in the document]
Comment (denton909) [reply __tests__/app.test.js:55]: I have updated this and test for each key.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -65,12 +65,12 @@ describe("nc-news-4", () => {
             expect(body.msg).toBe("Bad Request Invalid Input");
         })
         })
-        test('413 Request Entity Too Large. User inputs an id which is a number but is bigger than current amount of data this will return a 413 code and an error message', ()=>{
+        test('404 Request Not Found. User inputs an id which is a number but is bigger than current amount of data this will return a 404 code and an error message', ()=>{
             return request(app)
```
[the remark this answers is not in the document]
Comment (denton909) [reply __tests__/app.test.js:69]: Updated the test description to your suggestion.

**Pull request merged**